From 14598c446c46a6bf4330ba36dd356ea740d59a0b Mon Sep 17 00:00:00 2001
Message-ID: <14598c446c46a6bf4330ba36dd356ea740d59a0b.1705676149.git.mjg@fedoraproject.org>
From: Michael J Gruber <mjg@fedoraproject.org>
Date: Fri, 19 Jan 2024 15:53:59 +0100
Subject: [PATCH] fix time type on i686

Gcc14 turns some warnings into errors. Fix the time type error: mupdf
usues int64_t internally, system libs use time_t which coincide in most
architectures but not all of them.
---
 platform/gl/gl-main.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/platform/gl/gl-main.c b/platform/gl/gl-main.c
index b2eb0a48e..131b6dc83 100644
--- a/platform/gl/gl-main.c
+++ b/platform/gl/gl-main.c
@@ -2556,7 +2556,7 @@ static char *short_signature_error_desc(pdf_signature_error err)
 	}
 }
 
-const char *format_date(int64_t secs)
+const char *format_date(int64_t secs64)
 {
 	static char buf[100];
 #ifdef _POSIX_SOURCE
@@ -2564,6 +2564,7 @@ const char *format_date(int64_t secs)
 #else
 	struct tm *tm;
 #endif
+	time_t secs = secs64;
 
 	if (secs <= 0)
 		return NULL;
-- 
2.43.0.462.gcdfa2ea447

